cunumeric/_ufunc/ufunc.py

```diff
@@ -415,47 +433,41 @@ def __call__(
         # the dtype must be one of the dtypes supported by this operation.
         x, res_dtypes = self._resolve_dtype(x, precision_fixed)
 
-        if out is None:
-            results = tuple(
-                ndarray(shape=x.shape, dtype=res_dtype, inputs=(x, where))
-                for res_dtype in res_dtypes
-            )
-            out = results
-        else:
-
-            def maybe_create_output(out, res_dtype):
-                if out.dtype != res_dtype:
-                    if not np.can_cast(res_dtype, out.dtype, casting=casting):
-                        raise TypeError(
-                            f"Cannot cast ufunc '{self._name}' output from "
-                            f"{res_dtype} to {out.dtype} with casting rule "
-                            f"'{casting}'"
-                        )
-                    return ndarray(
-                        shape=out.shape, dtype=res_dtype, inputs=(x, where)
+        def maybe_create_output(out, res_dtype):
```

Comment:
This is not used anywhere?

Reply:
yeah I forgot to remove that function. will do it.

Reply:
297a5682d762acbe150023de8089f3659c3ec5da